packages/firebase_database/lib/ui/firebase_list.dart

```diff
@@ -47,9 +49,16 @@ class FirebaseList extends ListBase<DataSnapshot> with StreamSubscriberMixin<Eve
   /// Called when the data of the list has finished loading
   final ValueCallback onValue;
 
+  // ListBase implementation
   final List<DataSnapshot> _snapshots = <DataSnapshot>[];
   int get length => _snapshots.length;
+  set length(int value) {
+    _snapshots.length = value;
```

Comment:
This class is intended to provide an unmodifiable view of the list. I think the solution is to extend `UnmodifiableListBase` instead, or `throw new UnsupportedError("Cannot modify an unmodifiable list");`